src/Compilers/CSharp/Test/Symbol/Symbols/Source/FileModifierTests.cs

```diff
@@ -2,48 +2,2155 @@
+        Assert.Equal(c1.GetPublicSymbol(), info.Type);
+    }
+
+    [Fact]
+    public void Duplication_04()
+    {
+        var source1 = """
+            using System;
+
+            class C
+            {
+                public static void M()
+                {
+                    Console.Write(1);
+                }
+            }
+            """;
+
+        var main = """
+            using System;
+
+            file partial class C
+            {
+                public static void M()
+                {
+                    Console.Write(Number);
+                }
+            }
+
+            file partial class C
+            {
+                private static int Number => 2;
+            }
+
+            class Program
+            {
+                static void Main()
+                {
+                    C.M();
+                }
+            }
+            """;
+
+        var comp = CreateCompilation(new[] { source1, main }); // expectedOutput: 2
+        comp.VerifyDiagnostics();
+
+        var cs = comp.GetMembers("C");
+        Assert.Equal(2, cs.Length);
 
-            comp = CreateCompilation(source);
+        var c0 = cs[0];
+        Assert.True(c0 is SourceMemberContainerTypeSymbol { IsFile: false });
+        Assert.Equal(comp.SyntaxTrees[0], c0.DeclaringSyntaxReferences.Single().SyntaxTree);
+
+        var c1 = cs[1];
+        Assert.True(c1 is SourceMemberContainerTypeSymbol { IsFile: true });
+
+        var syntaxReferences = c1.DeclaringSyntaxReferences;
+        Assert.Equal(2, syntaxReferences.Length);
+        Assert.Equal(comp.SyntaxTrees[1], syntaxReferences[0].SyntaxTree);
+        Assert.Equal(comp.SyntaxTrees[1], syntaxReferences[1].SyntaxTree);
+
+        var tree = comp.SyntaxTrees[1];
+        var model = comp.GetSemanticModel(tree, ignoreAccessibility: true);
+        var cReference = tree.GetRoot().DescendantNodes().OfType<MemberAccessExpressionSyntax>().Last().Expression;
+        var info = model.GetTypeInfo(cReference);
+        Assert.Equal(c1.GetPublicSymbol(), info.Type);
+    }
+
+    [Theory]
+    [CombinatorialData]
+    public void Duplication_05(bool firstClassIsFile)
+    {
+        var source1 = $$"""
+            using System;
+
+            {{(firstClassIsFile ? "file " : "")}}partial class C
+            {
+                public static void M()
+                {
+                    Console.Write(1);
+                }
+            }
+            """;
+
+        var main = """
+            using System;
+
+            file partial class C
+            {
+                public static void M()
+                {
+                    Console.Write(2);
+                }
+            }
+
+            class Program
+            {
+                static void Main()
+                {
+                    C.M();
+                }
+            }
+            """;
+
+        var comp = CreateCompilation(new[] { source1, main }); // expectedOutput: 2
```

Comment:
>// expectedOutput: 2 [](http://example.com/codeflow?start=63&length=20)

I would have used a PROTOTYPE marker, but I think it's fine. Let's remember to hunt down every such instance